### PR TITLE
Generalize examples & more python code cleanup

### PR DESCRIPTION
**Pull request by danielfett**
In this PR, I generalized the examples: Right now, there is a lot of repetition in the code for each of the three examples. With more examples (W3C VC) and more example code (claims merging) to come, things will get more complicated.

I therefore moved the examples into a separate directory as YAML files. I used YAML, because this means that we can store all data required for each example as separate JSON documents in one YAML file (JSON is a subset of YAML). 

The code can now be called with a YAML file as the input and will run the demo for the example defined in the file. This will make it easier for us to add more examples in the future.

Also in this pull request:

- Cleaned up the code to produce just one example each time - there is no need to calculate all examples when only one is used in each run
- Cleaned up the code to produce both the CLI output and the spec examples from the same set of data to avoid repetitions
- This means that placeholders in main.md are now strictly named as follows: `example-{example_id}-{artifact_id}`, where `artifact_id` refers to the identifier of the data produced in the example run (see `sd_jwt` file).
- Fix for example output: serialized SVC was not shown, but serialized SD-JWT twice
- Renamed `_sel_exp` to `_artifacts` (not sure what `_sel_exp` was meant to say)
- Fix for no_randomness: RNG was seeded to static value even when no_randomness was not used

### CODE REVIEW

**Comment by peppelinux** [comment]
> Cleaned up the code to produce just one example each time - there is no need to calculate all examples when only one is used in each run

Yes, it was in my backlog, thanx. 
This first proposal was to get all the examples in a single shot, we can add a CLI parameter, I'll have a look for that

> Cleaned up the code to produce both the CLI output and the spec examples from the same set of data to avoid repetitions
This means that placeholders in main.md are now strictly named as follows: example-{example_id}-{artifact_id}, where artifact_id refers to the identifier of the data produced in the example run (see sd_jwt file).

Thank you, I wanted to make a proposal for this, you did it

> Fix for example output: serialized SVC was not shown, but serialized SD-JWT twice

My bad

> Renamed _sel_exp to _artifacts (not sure what _sel_exp was meant to say)

it was "selected example" but artifacts is better anyway

> Fix for no_randomness: RNG was seeded to static value even when no_randomness was not used

OK!

File: sd_jwt/bin/sd_jwt [hunk 1]
```diff
@@ -23,325 +22,235 @@ logger = logging.getLogger("sd_jwt")
...
-    sys.exit(f'{__version__}')
+    sys.exit(f"{__version__}")
+
+### Load settings
 
-# load settings
 if _args.settings_path:
     sys.path.append(_args.settings_path)
-    from settings import *
+    from settings import (
```
Comment (peppelinux) [inline sd_jwt/bin/sd_jwt:106]: * prevented code repetition but this solution is fair

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: sd_jwt/bin/sd_jwt (hunk 1 above)
Comment (danielfett) [reply sd_jwt/bin/sd_jwt:106]: Keeping as proposed is okay?

File: sd_jwt/bin/sd_jwt [hunk 2]
```diff
@@ -23,325 +22,235 @@ logger = logging.getLogger("sd_jwt")
...
+
+sd_jwt_release_payload, serialized_sd_jwt_release = create_release_jwt(
+    _args.nonce,
+    VERIFIER,
+    example["disclosed_claims"],
+    _artifacts["serialized_svc"][0],
+    HOLDER_KEY,
+)
+
+_combined = _artifacts["serialized_sd_jwt"][0] + "." + serialized_sd_jwt_release
```
[the remark this answers is not in the document]
Comment (danielfett) [reply sd_jwt/bin/sd_jwt:195]: Intentional decision - I found this much cleaner.

File: sd_jwt/examples/complex.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+user_claims:
```
[the remark this answers is not in the document]
Comment (danielfett) [reply sd_jwt/examples/complex.yml:1]: As I wrote, JSON is a subset of YAML. The file is a YAML file and the file ending is correct, but we can store the user details as JSON which is close to what we show in the spec. I chose not to use JSON for the full file, as one big JSON structure would look convoluted.

**Review by Sakurann — APPROVED** [review]

**Pull request merged**